### PR TITLE
test: add governance proposal validator fuzz target

### PR DESCRIPTION
**Pull request by thepastaclaw**
## Summary
- add a new Dash-specific functional fuzz target: `governance_proposal_validator`
- exercise `CProposalValidator` with both structured and random proposal payloads
- cover key validator pathways including:
  - JSON parsing failures and malformed payloads
  - oversized data handling (`MAX_DATA_SIZE` path)
  - script vs non-script payment addresses (`fAllowScript` toggle)
  - expiration toggles (`fCheckExpiration` true/false)
  - URL edge cases including bracket mismatch checks
- wire the new target into `src/Makefile.test.include`

## Testing
- `./configure --enable-fuzz-binary --disable-tests --without-gui --disable-bench --disable-stacktraces`
- `make -C src test/fuzz/fuzz -j8`

Tracker: https://github.com/thepastaclaw/tracker/issues/129  
Epic: https://github.com/thepastaclaw/tracker/issues/108

## Validation

**What was tested:**
- `./configure --enable-fuzz-binary --disable-tests --without-gui --disable-bench --disable-stacktraces` + `make -C src test/fuzz/fuzz -j8` — built the fuzz binary including the new `governance_proposal_validator` target
- Ran target with empty corpus to confirm it starts up and exercises the validator paths without immediate crashes

**Results:**
- `governance_proposal_validator` fuzz target compiled and linked successfully
- Target ran for a short seed corpus pass with no crashes or sanitizer errors
- CI full suite: all checks passed
  - `linux64-build` — pass; `linux64-test` — pass
  - `linux64_nowallet-build` — pass; `linux64_nowallet-test` — pass
  - `linux64_sqlite-build` — pass; `linux64_sqlite-test` — pass
  - `linux64_tsan-build` — pass; `linux64_tsan-test` — pass
  - `linux64_ubsan-build` — pass; `linux64_ubsan-test` — pass
  - `mac-build / Build source` — pass
  - `win64-build / Build source` — pass

**Environment:** Local macOS arm64 (fuzz build + empty corpus run); GitHub Actions CI (linux64 + tsan + ubsan + mac + win64 matrix)

### CODE REVIEW

File: test/util/data/non-backported.txt [hunk 1]
```diff
@@ -50,7 +50,7 @@ src/stacktraces.*
 src/stats/*.cpp
 src/stats/*.h
 src/test/block_reward_reallocation_tests.cpp
-src/test/bls_tests.cpp
+src/test/fuzz/governance_proposal_validator.cppsrc/test/bls_tests.cpp
```
Comment (PastaPastaPasta) [inline test/util/data/non-backported.txt]: seems you have some new lines missed here
